File: Stack/QueueFromStack/cpp/code.cpp
#include <bits/stdc++.h>
using namespace std;

class Queue {
public:
	stack<int> s1;
	stack<int> s2;

	void enQueue(int x) {
		s1.push(x);
	}

	int deQueue() {
		if(s1.empty() && s2.empty()) {
			cout << "Queue is empty" << endl;
			exit(0);
		}

		if(s2.empty()) {
			while(!s1.empty()) {
				s2.push(s1.top());
				s1.pop();
			}
		}

		int x = s2.top();
		s2.pop();
		return x;
	}
};

int main() {

	Queue q;

	q.enQueue(1);
	q.enQueue(2);
	q.enQueue(3);

	for(int i=0; i<3; i++) {
		cout  << q.deQueue() << " ";
	}
	cout << endl;

	return 0;
}

File: Stack/QueueFromStack/java/code.java
import java.util.Stack;

class Code {

	static class Queue {
		Stack<Integer> s1;
		Stack<Integer> s2;

		Queue() {
			s1 = new Stack<Integer>();
			s2 = new Stack<Integer>();
		}
	}

	static void enQueue(Queue q, int x) {
		q.s1.push(x);
	}

	static int deQueue(Queue q) {
		int x;

		if(q.s1.isEmpty() && q.s2.isEmpty()) {
			System.out.println("Q is Empty");
			System.exit(0);
		}

		if(q.s2.isEmpty()) {
			while(!q.s1.isEmpty()) {
				q.s2.push(q.s1.pop());
			}
		}

		return q.s2.pop();
	}

	public static void main(String[] args) {

		Queue q = new Queue();

		enQueue(q, 1);
		enQueue(q, 2);
		enQueue(q, 3);

		for(int i=0; i<3; i++) {
			System.out.println(deQueue(q));
		}
		
	}
}